### PR TITLE
Get the total number of messages of a thread for `BP_Messages_Thread`

### PR DESCRIPTION
**Pull request by renatonascalves**
Trac ticket: https://buddypress.trac.wordpress.org/ticket/8697

---
**This Pull Request is for code review only. Please keep all other discussion in the BuddyPress Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the WordPress Core Handbook for more details.**

### CODE REVIEW

**Comment by renatonascalves** [comment]
We could also not take this approach at all and do the actual MySQL query in the BP REST API plugin. Here I'm assuming this new piece of data might be useful for those using or extending `BP_Messages_Thread`.

File: src/bp-messages/classes/class-bp-messages-thread.php [hunk 1]
```diff
@@ -945,6 +956,43 @@ public static function get_total_threads_for_user( $user_id, $box = 'inbox', $ty
...
+	 * @global BuddyPress $bp The one true BuddyPress instance.
+	 * @global wpdb $wpdb WordPress database object.
+	 *
+	 * @param int $thread_id The message thread ID.
+	 * @return int Total thread message count
+	 */
+	public static function get_total_thread_message_count( $thread_id ) {
+		global $wpdb;
+
+		$total_count = wp_cache_get( $thread_id, 'bp_messages_unread_count' );
```
Comment (imath) [inline src/bp-messages/classes/class-bp-messages-thread.php:973]: I think we should use another cache group, as this one is used to get the unread count. This function, if I understand well, gets the amount of messages for a thread no matter if messages are unread or read. About the cache group name, something like `bp_messages_thread_count` might be suitable...

NB: You'll need to also add the new cache group to the `BP_Messages_Component::setup_cache_groups()`

**Review by imath — CHANGES_REQUESTED** [review]
Thanks for working on this. I have a concern about how the count is cached. We should use a specific cache group instead of one being used for something else (get unread messages count). If you can add unit tests about cache, that would be terrific 😄

File: src/bp-messages/classes/class-bp-messages-thread.php (hunk 1 above)
Comment (renatonascalves) [reply src/bp-messages/classes/class-bp-messages-thread.php:973]: Ops! Good cache! This is a mistake!

**Comment by renatonascalves** [comment]
@imath Thanks for the cache reminder. I updated the code to use a different approach, including support for cache busting. :)

**Review by imath — COMMENTED** [review]
Awesome 👏. Ready to be committed on SVN trunk 👌

**Pull request closed without merging**